Sdk/MoyasarSdk/Models/PaymentRequest.swift

```diff
@@ -25,6 +24,29 @@ public struct PaymentRequest {
         self.allowedNetworks = allowedNetworks
     }
     
+    public init(
```

Comment:
- apiKey validation already exists 
- allowedNetworks  already has a default value 
- metadata user can send String, Integer, Float or Boolean value wrapped in string to avoid the complex unneeded check for each type and never confuse user 
- description ---> Done
- buttonType --->  what you think if we set it with default text as Pay and let user if he want to change the text to pass the button title to avoid  all this localization  to let user determine the button text? 


<img width="464" alt="Screenshot 2024-09-30 at 5 03 16 PM" src="https://github.com/user-attachments/assets/e44408e7-268e-4d3f-92e7-49d28da34b44">

Reply:
Hey @Mahmoud-Abdelwahab 

* According to the requirements, the validation for `apiKey` should be under the `PaymentRequest` struct and not within the `PaymentService`. Can you please relocate the validation?
* The current `allowedNetworks` default value contain all the networks. However, the requirement asks to have only  `mada`, `visa` and `mastercard` as the default values. Can you please do the change?
* For the `metadata`, I see where you are coming from. However, supporting different types, including String, improve the ergonomics of the API and is an appreciated feature by our customer. Our API already pre-fill some of the metadata on behalf of the customer using Boolean and Integer types to deliver certain features. I hope you are convinced now. 👌
* Regarding `buttonType`, I suggest including all these different translations within the current i18n system. The client should only pass the pre-defined key to get the desired text according to the set localization config. This reflects better consistency across all the SDK's localization features and prevents from having some text in a language and other text in "any" language. What do you think?

Reply:
Done bro 
Regarding   i18n system. it very good idea but i think this will cost fee and  this is a small-scale projects we can unify them by using the same localization as written in notion  and i did that for now